### DISCUSSION TITLE
websocket reconnection

### DISCUSSION THREAD

**Discussion by xingsongs** [actix/actix-web#3110] [General] [answered]
Developed with actix_web_actors, when the webscoket is disconnected, do not close the actor. When the websocket request arrives, according to the identification such as the ID of the url, the previous actor is matched, and then a StreamHandler is started。

God please give me some advice。

**Comment by dns2utf8-tipi** [comment, marked as answer]
You would have to run an `Arena { active_sessions: Vec<Addr<MySession>> }` that all WebSockets connect to. 

1) If the client is new: give it a random secret string. eg. 32 random Bytes to hold on to and store it in `MySession` as well
2) The WebSocket disconnects, but the `MySession` keeps itself alive with a timer, eg. for 5min
3) If the client reconnects within that time, it sends the random secret and the WebSocket actor can find the `MySession` with the secret or not if it is too late

**Reply by xingsongs** [reply]
thank you very much for your solution.
